portal-impl/src/com/liferay/portal/servlet/filters/language/LanguageFilter.java

```diff
@@ -75,8 +76,17 @@ protected void processFilter(
 			HttpServletResponse httpServletResponse, FilterChain filterChain)
 		throws Exception {
 
-		httpServletResponse.setHeader(
-			HttpHeaders.CACHE_CONTROL, "private, no-cache");
+		String queryString = httpServletRequest.getQueryString();
```

Comment:
:thinking: Why don't you use [ServletRequest.getParameter](https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getParameter(java.lang.String))?